### DISCUSSION TITLE
Check input and pattern target mismatch

### DISCUSSION THREAD

**Discussion by liutiming** [ropensci/targets#538] [Ideas]
Hi Will, this is just a suggestion as I am not sure about others' use cases. 

I often find that the input target and pattern target are the same; often I also produce bugs when I copy code and forget to change the pattern target name. Thus I wonder what your thoughts are on implementing a `{targets}` level check on whether these two targets match or not, and give warning if not

**Comment by wlandau** [comment]
Are you talking about something like this?

```r
# _targets.R
library(targets)
list(
  tar_target(x, seq_len(2)),
  tar_target(y, x, pattern = map(y)) # typo; should be map(x)
)
```

`tar_make()` etc. do throw an error (although currently cryptic).

```r
tar_make()
#> Error : "y" is in "y"
```

Were you thinking of some other kind of user mistake?

**Reply by liutiming** [reply]
kind of like this

```
# _targets.R
library(targets)
list(
  tar_target(x, seq_len(2)),
  tar_target(y, x, pattern = map(x)),
  tar_target(z, y, pattern = map(x)) # typo; should be map(y), because I copied pasted the second line
)
```

**Reply by wlandau** [reply]
Sometimes, `tar_target(z, y, pattern = map(x))` is actually correct: for example, if the dynamic branches are independent draws from a distribution and `x` is the rep index. `targets` cannot infer enough context to tell.

**Reply by liutiming** [reply]
Yes I understand that. I am just wondering from a design perspective which is the more common option and whether the the software should adapt to the more common use case...

**Reply by wlandau** [reply]
Both situations are common enough that a message or warning would be superfluous most of the time. `tarchetypes::tar_map()`, several functions in [`stantargets`](https://docs.ropensci.org/stantargets/), and several functions in [`jagstargets`](https://wlandau.github.io/jagstargets) make heavy use of `tar_target(z, y, pattern = map(x))`. At my work, for example, I spend a lot of time validating Stan models using the technique at https://docs.ropensci.org/stantargets/articles/simulation.html.

**Reply by liutiming** [reply]
I see. Thanks a lot for the kind response!
